File: src/main/java/com/braintreegateway/TransactionOptionsRequest.java
package com.braintreegateway;

public class TransactionOptionsRequest extends BaseRequest {
    private Boolean addBillingAddressToPaymentMethod;
    private Boolean holdInEscrow;
    private TransactionRequest parent;
    private Boolean storeInVault;
    private Boolean storeInVaultOnSuccess;
    private Boolean storeShippingAddressInVault;
    private Boolean submitForSettlement;
    private String venmoSdkSession;

    public TransactionOptionsRequest(TransactionRequest parent) {
        this.parent = parent;
    }

    public TransactionOptionsRequest addBillingAddressToPaymentMethod(Boolean addBillingAddressToPaymentMethod) {
        this.addBillingAddressToPaymentMethod = addBillingAddressToPaymentMethod;
        return this;
    }

    public TransactionRequest done() {
        return parent;
    }

    public TransactionOptionsRequest holdInEscrow(Boolean holdInEscrow) {
        this.holdInEscrow = holdInEscrow;
        return this;
    }

    public TransactionOptionsRequest storeInVault(Boolean storeInVault) {
        this.storeInVault = storeInVault;
        return this;
    }

    public TransactionOptionsRequest storeInVaultOnSuccess(Boolean storeInVaultOnSuccess) {
        this.storeInVaultOnSuccess = storeInVaultOnSuccess;
        return this;
    }

    public TransactionOptionsRequest storeShippingAddressInVault(Boolean storeShippingAddressInVault) {
        this.storeShippingAddressInVault = storeShippingAddressInVault;
        return this;
    }

    public TransactionOptionsRequest submitForSettlement(Boolean submitForSettlement) {
        this.submitForSettlement = submitForSettlement;
        return this;
    }

    public TransactionOptionsRequest venmoSdkSession(String venmoSdkSession) {
        this.venmoSdkSession = venmoSdkSession;
        return this;
    }

    @Override
    public String toXML() {
        return buildRequest("options").toXML();
    }

    @Override
    public String toQueryString() {
        return toQueryString("options");
    }

    @Override
    public String toQueryString(String root) {
        return buildRequest(root).toQueryString();
    }

    protected RequestBuilder buildRequest(String root) {
        return new RequestBuilder(root).
            addElement("holdInEscrow", holdInEscrow).
            addElement("storeInVault", storeInVault).
            addElement("storeInVaultOnSuccess", storeInVaultOnSuccess).
            addElement("addBillingAddressToPaymentMethod", addBillingAddressToPaymentMethod).
            addElement("storeShippingAddressInVault", storeShippingAddressInVault).
            addElement("submitForSettlement", submitForSettlement).
            addElement("venmoSdkSession", venmoSdkSession);
    }
}

File: src/main/scala/com/braintreegateway/TransactionOptionsRequest.scala
package com.braintreegateway

class TransactionOptionsRequest(val done: TransactionRequest) extends BaseRequest with HasParent[TransactionRequest] {

  def addBillingAddressToPaymentMethod(addBillingAddressToPaymentMethod: Boolean): TransactionOptionsRequest = {
    this.addBillingAddressToPaymentMethod = addBillingAddressToPaymentMethod
    this
  }

  def holdInEscrow(holdInEscrow: Boolean): TransactionOptionsRequest = {
    this.holdInEscrow = holdInEscrow
    this
  }

  def storeInVault(storeInVault: Boolean): TransactionOptionsRequest = {
    this.storeInVault = storeInVault
    this
  }

  def storeInVaultOnSuccess(storeInVaultOnSuccess: Boolean): TransactionOptionsRequest = {
    this.storeInVaultOnSuccess = storeInVaultOnSuccess
    this
  }

  def storeShippingAddressInVault(storeShippingAddressInVault: Boolean): TransactionOptionsRequest = {
    this.storeShippingAddressInVault = storeShippingAddressInVault
    this
  }

  def submitForSettlement(submitForSettlement: Boolean): TransactionOptionsRequest = {
    this.submitForSettlement = submitForSettlement
    this
  }

  def venmoSdkSession(venmoSdkSession: String): TransactionOptionsRequest = {
    this.venmoSdkSession = venmoSdkSession
    this
  }

  override def toXmlString: String = {
    buildRequest("options").toXmlString
  }

  override def toQueryString: String = {
    toQueryString("options")
  }

  override def toQueryString(root: String): String = {
    buildRequest(root).toQueryString
  }

  protected def buildRequest(root: String): RequestBuilder = {
    new RequestBuilder(root).
      addElement("holdInEscrow", holdInEscrow).
      addElement("storeInVault", storeInVault).
      addElement("storeInVaultOnSuccess", storeInVaultOnSuccess).
      addElement("addBillingAddressToPaymentMethod", addBillingAddressToPaymentMethod).
      addElement("storeShippingAddressInVault", storeShippingAddressInVault).
      addElement("submitForSettlement", submitForSettlement).
      addElement("venmoSdkSession", venmoSdkSession)
  }

  private var addBillingAddressToPaymentMethod: java.lang.Boolean = null
  private var holdInEscrow: java.lang.Boolean = null
  private var storeInVault: java.lang.Boolean = null
  private var storeInVaultOnSuccess: java.lang.Boolean = null
  private var storeShippingAddressInVault: java.lang.Boolean = null
  private var submitForSettlement: java.lang.Boolean = null
  private var venmoSdkSession: String = null
}